examples/plot_intelex.py

```diff
@@ -0,0 +1,276 @@
+print(
+    f"[stock scikit-learn] Inference took t_stock = {t_stock:.2f}s with a "
+    f"log-loss score of {log_loss_score:.1%}"
+)
+
+start = perf_counter()
+y_proba = clf_opt.predict_proba(X_test)
+t_opt = perf_counter() - start
+
+log_loss_score = log_loss(y_test, y_proba)
+print(
+    f"[sklearnex] Inference took t_opt = {t_opt:.2f}s with a log-loss score of"
+    f" {log_loss_score:.1%}"
+)
+
+print(f"t_stock / t_opt = {t_stock/t_opt:.1f}")
+
+# %%
+# We see that inference using ``sklearnex`` is a lot faster while achieving the
+# same log-loss score. In fact, the agreement between stock Scikit-learn and
+# ``sklearnex`` model outputs is guaranteed by CI tests, see
+# `here <https://github.com/intel/scikit-learn-intelex/blob/master/tests/test_examples.py>`_.
+
+# %%
+# Save and upload the models
+# ==========================
+# Let's save all required files to disk and initialize Hugging Face Model Hub
+# repositories.
+
+with open("model-stock.pkl", mode="bw") as fp:
+    pickle.dump(clf, file=fp)
+
+with open("model-optimized.pkl", mode="bw") as fp:
+    pickle.dump(clf_opt, file=fp)
+
+# replace with your own token or set it as an environment variable before
+# running the script
+token = os.environ["HF_HUB_TOKEN"]
+
+# set up two repos, one for each model
+stock_repo = mkdtemp(prefix="stock-")
+hub_utils.init(
+    model="model-stock.pkl",
+    requirements=["scikit-learn=1.2.1"],
+    dst=stock_repo,
+    task="tabular-classification",
+    data=X_test,
+)
+
+opt_repo = mkdtemp(prefix="opt-")
+hub_utils.init(
+    model="model-optimized.pkl",
+    requirements=["scikit-learn=1.2.1", "scikit-learn-intelex=2023.0.1"],
+    dst=opt_repo,
+    task="tabular-classification",
+    data=X_test,
+    use_intelex=True,
+)
+
+# Create Model cards with the most basic information
+clf_card = card.Card(clf, metadata=card.metadata_from_config(Path(stock_repo)))
+clf_card.save(Path(stock_repo) / "README.md")
+
+clf_opt_card = card.Card(clf_opt, metadata=card.metadata_from_config(Path(opt_repo)))
+clf_opt_card.save(Path(opt_repo) / "README.md")
+
+# Push everything to the Model hub
+user_name = whoami(token=token)["name"]
+uuid = uuid4()
+repo_id_stock = f"{user_name}/knn-example-stock-{uuid}"
+repo_id_opt = f"{user_name}/knn-example-intelex-{uuid}"
+
+print(f"Pushing skl model to: {repo_id_stock}")
+hub_utils.push(
+    repo_id=repo_id_stock,
+    source=stock_repo,
+    token=token,
+    commit_message="Add scikit-learn KNN model example",
+    create_remote=True,
+    private=False,
+)
+print(f"Pushing sklearnex model to: {repo_id_opt}")
+hub_utils.push(
+    repo_id=repo_id_opt,
+    source=opt_repo,
+    token=token,
+    commit_message="Add scikit-learn-intelex KNN model example",
+    create_remote=True,
+    private=False,
+)
+
+# %%
+# Download and use the models from Model Hub
+# ==========================================
+# We download the models again and check whether our achieved performance boost
+# survived the round trip to the Hub.
```

Comment:
do we need to test this? I think this is well tested in other places

Reply:
Wanted to demonstrate that we can download and use optimized models. Should I remove it?

Reply:
I don't mind, as you wish. I'm not sure if it's adding to the message here though.

Reply:
removed